### PR TITLE
[chore] Fix daemonset-features E2E test for OpenShift

### LINKED ISSUE #1353: [bug] daemonset-features E2E test doesn't pass in OpenShift because some steps are missing
The `daemonset-features` E2E test cannot pass in OpenShift because some extra steps are needed in order to deploy a DaemonSet in OpenShift.

## Steps to reproduce
1. Build and deploy the OpenTelemetry Operator in an OpenShift Cluster (OCP 4.11 in my case): `$ make prepare-e2e`
2. Run the `daemonset-features` E2E test: `$ kuttl test --test daemonset-features`

As result, the test will not pass.

### Details of the issue
When this CR of an OpenTelemetry Collector is deployed:
```yaml
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: daemonset
spec:
  mode: daemonset
  hostNetwork: true
  config: |
    receivers:
      jaeger:
        protocols:
          grpc:
    processors:
    exporters:
      logging:
    service:
      pipelines:
        traces:
          receivers: [jaeger]
          processors: []
          exporters: [logging]
```

The pods are not started. The description of the Daemonset (`$ kubectl describe daemonset daemonset-collector`) shows the following warning:
```
 Warning  FailedCreate  9s (x13 over 30s)  daemonset-controller  Error creating: pods "daemonset-collector-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, spec.containers[0].securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, spec.containers[0].securityContext.containers[0].hostPort: Invalid value: 14250: Host ports are not allowed to be used, spec.containers[0].securityContext.containers[0].hostPort: Invalid value: 8888: Host ports are not allowed to be used, provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or service
```

## How to fix it:
1. Create a Service Account (`otel-collector-daemonset` for example) and a Security Context Constrain allowing the usage of the HostPort and HostNetworks:
```yaml
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: daemonset-with-hostport
  annotations:
    kubernetes.io/description: 'Allows DaemonSets to bind to a well-known host port'
runAsUser:
  type: RunAsAny
seLinuxContext:
  type: RunAsAny
allowHostPorts: true
allowHostDirVolumePlugin: false
allowHostIPC: false
allowHostNetwork: true
allowHostPID: false
allowPrivilegedContainer: false
readOnlyRootFilesystem: false
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: otel-collector-daemonset
```
2. Add the SCC to the Service Account: `$ oc adm policy add-scc-to-user -z otel-collector-daemonset daemonset-with-hostport -n $NAMESPACE`
3. And set the `.spec.serviceAccount` from the OpenTelemetry Collector (patching the previous OTEL Collector, for instance `$ kubectl patch otelcol daemonset --type=merge -p '{"spec":{"serviceAccount":"otel-collector-daemonset"}}'`)

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancasa@gmail.com>

Fixes #1353

### CODE REVIEW

File: tests/e2e/daemonset-features/scc.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+kind: SecurityContextConstraints
```
Comment (pavolloffay) [inline tests/e2e/daemonset-features/scc.yaml:1]: It's similar to what jaeger-operator does https://github.com/jaegertracing/jaeger-operator/blob/dcb7d7f4bec09ab74b17cac799c3502c6b314158/examples/openshift/hostport-scc-daemonset.yaml

File: tests/e2e/daemonset-features/add-sa-collector.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+if [[ "$(kubectl api-resources)" =~ "operator.openshift.io" ]]; then
```
Comment (pavolloffay) [inline tests/e2e/daemonset-features/add-sa-collector.sh:2]: might be better `kubectl api-resources --api-group=operator.openshift.io  -o name` and check if there is any output

File: tests/e2e/daemonset-features/add-scc-openshift.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+if [[ "$(kubectl api-resources)" =~ "operator.openshift.io" ]]; then
```
Comment (pavolloffay) [inline tests/e2e/daemonset-features/add-scc-openshift.sh:2]: The same here `kubectl api-resources --api-group=operator.openshift.io  -o name`

**Review by pavolloffay — COMMENTED** [review]
LGTM just some nits

File: tests/e2e/daemonset-features/scc.yaml (hunk 1 above)
Comment (iblancasa) [reply tests/e2e/daemonset-features/scc.yaml:1]: Yes. I copied, pasted and adapted.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**